devito/dle/parallelizer.py

```diff
@@ -161,8 +159,10 @@ def _make_partree(self, candidates, omp_pragma=None):
                 if i.is_Vectorizable:
                     break
 
+                # Would there be enough work per parallel iteration?
                 try:
-                    work = reduce(mul, [int(j.dim.symbolic_size) for j in candidates[n:]])
+                    work = prod([int(j.dim.symbolic_size) for j in candidates[n+1:]],
+                                default=Ompizer.COLLAPSE_WORK)
```

Comment:
I don't think you want this default here - the last argument to `reduce` is prepended to the list, which means this will never be less than `Ompizer.COLLAPSE_WORK`.  I think we need to handle the empty list case explicitly.

```
In [4]: reduce(mul, [1,2,3],2)
Out[4]: 12
```

Reply:
Actually, what's the harm in it returning `1` if there are no deeper loops?  Isn't that exactly the truth? (and indeed, we would not want it to include that last non-loop in the collapse).